### ISSUE TITLE
How can I solve the problem: cannot open shared object file: No such file or directory No plugin found (libnccl-net.so), using internal implementation

### ISSUE THREAD

**Issue by Leon1207** [NVIDIA/nccl#1231]
Error info bellow:

torch.distributed.DistBackendError: NCCL error in: ../torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:1691, unhandled cuda error (run with NCCL_DEBUG=INFO for details), NCCL version 2.19.3
ncclUnhandledCudaError: Call to CUDA function failed.
Last error:
Cuda failure 802 'system not yet initialized'

NCCL_DEBUG=INFO for details:

VM-0-14-tencentos:162637:162637 [6] NCCL INFO Bootstrap : Using eth0:172.17.0.14<0>
VM-0-14-tencentos:162637:162637 [6] NCCL INFO NET/Plugin : dlerror=libnccl-net.so: cannot open shared object file: No such file or directory No plugin found (libnccl-net.so), using internal implementation
VM-0-14-tencentos:162635:162635 [4] NCCL INFO cudaDriverVersion 12020

**Comment by AddyLaddy** [comment]
This message 

`NET/Plugin : dlerror=libnccl-net.so: cannot open shared object file: No such file or directory No plugin found (libnccl-net.so), using internal implementation`

is harmless. It just means that NCCL did not find an external network plugin (e.g. to support IB/SHARP).
It defaults back to its internal plugin, which supports IB Verbs, RoCE and Sockets.

**Comment by Leon1207** [comment]
Thank you very much. And can I ask why this error occurred?

torch.distributed.DistBackendError: NCCL error in: ../torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:1691, unhandled cuda error (run with NCCL_DEBUG=INFO for details), NCCL version 2.19.3
ncclUnhandledCudaError: Call to CUDA function failed.
Last error:
Cuda failure 802 'system not yet initialized'

Do you have any advice?

**Comment by AddyLaddy** [comment]
It looks like CUDA isn't working at all on your system. I'd download and run something like [nvbandwidth](https://github.com/NVIDIA/nvbandwidth)
to check basic CUDA and inter GPU communication is functional.

**Comment by Leon1207** [comment]
Thanks for your advice! I will try it.

**Comment by TDW1114** [comment]
> This message
> 
> `NET/Plugin : dlerror=libnccl-net.so: cannot open shared object file: No such file or directory No plugin found (libnccl-net.so), using internal implementation`
> 
> is harmless. It just means that NCCL did not find an external network plugin (e.g. to support IB/SHARP). It defaults back to its internal plugin, which supports IB Verbs, RoCE and Sockets.

what if I run multinode training? It fails to run due to this problem I think

**Comment by sjeaugey** [comment]
> what if I run multinode training? It fails to run due to this problem I think

No, it doesn't. It just means that we're not using an external network plugin and use the internal Socket or IB/RoCE support instead.

If a run fails, check the log and look for `NCCL WARN`. `NCCL INFO` lines are just information.

**Comment by maherr13** [comment]
I'm using the multinode training case, and i faced the same problem above that it didnt connect to the external network and back to the internal. 

here is the INFO 
```
NCCL INFO NET/Plugin : dlerror=libnccl-net.so: cannot open shared object file: No such file or directory No plugin found (libnccl-net.so), using internal implementation
GPU-15:77070:77070 [0] NCCL INFO cudaDriverVersion 12010
```

here is the WARN
```
NCCL WARN socketStartConnect: Connect to 172.18.0.1<58897> failed : Software caused connection abort
```

and Here is the ERROR
```
Error executing method `init_device`. This might cause deadlock in distributed execution.
```

what should i do in this case ?
